### PR TITLE
Pawelka/0x1e

### LINKED ISSUE #550: message is incomplete
The sever called the client with a complex object,and the browser's console log the data received correct. However, an error throws after the log:uncaught error:message is incomplete.

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes #550

### CODE REVIEW

**Comment by moozzyk** [comment]
Rebased on latest dev

File: src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageWriter.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using System.Buffers;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.AspNetCore.SignalR.Internal.Encoders
+{
+    public static class LengthPrefixedTextMessageWriter
+    {
+        private const int Int32OverflowLength = 10;
```
Comment (muratg) [inline src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageWriter.cs]: Is this the number of decimal digits that when Int32 has less, there's no overflow?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageWriter.cs]: This is the max number of digits an Int32 number can have. (btw. this is not new code - it's the old TextMessageParser moved here and renamed, same with LengthPrefixedTextMessageParser)

File: src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageWriter.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageWriter.cs]: This was previously `TextMessageFormatter`

File: src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageParser.cs [hunk 3]
```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.SignalR.Common/Internal/Encoders/LengthPrefixedTextMessageParser.cs]: This was previously TextMessageParser

**Pull request merged**